src/plugins/intel_cpu/src/emitters/snippets/x64/jit_memory_emitters.cpp

```diff
@@ -168,51 +151,41 @@ void jit_store_memory_emitter::emit_impl(const std::vector<size_t>& in,
 }
 
 template <cpu_isa_t isa>
-void jit_store_memory_emitter::emit_isa(const std::vector<size_t> &in, const std::vector<size_t> &out) const {
-    if (!store_emitter)
-        OV_CPU_JIT_EMITTER_THROW("Store CPU emitter isn't initialized!");
-    store_emitter->emit_code({in[0], byte_offset}, {out[0]}, aux_vec_idxs, aux_gpr_idxs);
-}
+void jit_load_broadcast_emitter::emit_isa(const std::vector<size_t> &in, const std::vector<size_t> &out) const {
+    using Vmm = typename dnnl::impl::utils::conditional3<isa == dnnl::impl::cpu::x64::sse41,
+            Xmm, isa == dnnl::impl::cpu::x64::avx2, Ymm, Zmm>::type;
+    Reg64 in_reg(in[0]);
+    Vmm vmm_dst = Vmm(out[0]);
 
-void jit_store_memory_emitter::emit_data() const {
-    store_emitter->emit_data();
+    // In doesn't really matter if we broadcast or `movss` for vector tails so keep only one version for `BroadcastLoad`,
+    // key point here is not to add post-increment, it might be fixed by some other approach in future
+    switch (src_prc.size()) {
+        case 4: h->uni_vbroadcastss(vmm_dst, h->ptr[in_reg + compiled_byte_offset]); break;
+        case 2: h->vpbroadcastw(vmm_dst, h->ptr[in_reg + compiled_byte_offset]); break;
+        case 1: h->vpbroadcastb(vmm_dst, h->ptr[in_reg + compiled_byte_offset]); break;
+        default: OV_CPU_JIT_EMITTER_THROW("Unsupported data type");
+    }
 }
 
-jit_store_convert_emitter::jit_store_convert_emitter(jit_generator* h, cpu_isa_t isa, const ExpressionPtr& expr)
-    : jit_memory_emitter(h, isa, expr) {
-    const auto store = ov::as_type_ptr<snippets::op::Store>(expr->get_node());
-    count = store->get_count();
-    byte_offset = store->get_offset();
-    in_out_type_ = emitter_in_out_map::vec_to_gpr;
-
+jit_store_memory_emitter::jit_store_memory_emitter(jit_generator* h, cpu_isa_t isa, const ExpressionPtr& expr)
+    : jit_memory_emitter(h, isa, expr, emitter_in_out_map::vec_to_gpr) {
     if (ov::is_type<ov::intel_cpu::StoreConvertTruncation>(expr->get_node())) {
         store_emitter.reset(new jit_store_emitter(h, isa, src_prc, dst_prc, count, arithmetic_mode::truncation));
     } else if (ov::is_type<ov::intel_cpu::StoreConvertSaturation>(expr->get_node())) {
         store_emitter.reset(new jit_store_emitter(h, isa, src_prc, dst_prc, count, arithmetic_mode::saturation));
-    }
-}
-
-void jit_store_convert_emitter::emit_impl(const std::vector<size_t>& in,
-                                    const std::vector<size_t>& out) const {
-    if (host_isa_ == dnnl::impl::cpu::x64::sse41) {
-        emit_isa<dnnl::impl::cpu::x64::sse41>(in, out);
-    } else if (host_isa_ == dnnl::impl::cpu::x64::avx2) {
-        emit_isa<dnnl::impl::cpu::x64::avx2>(in, out);
-    } else if (host_isa_ == dnnl::impl::cpu::x64::avx512_core) {
-        emit_isa<dnnl::impl::cpu::x64::avx512_core>(in, out);
+    } else if (ov::is_type<ov::snippets::op::Store>(expr->get_node())) {
+        store_emitter.reset(new jit_store_emitter(h, isa, src_prc, dst_prc, count));
```

Comment:
Minor: it seems like we create the same `jit_store_emitter` for `StoreConvertSaturation` and `Store` operations. So we could make one `else` branch instead of 2 `else if` ones.
And to keep the existing node type check (which is in `else` branch now) we could just add the following assert:
`OV_CPU_JIT_EMITTER_ASSERT(ov::is_type<ov::snippets::op::Store>(expr->get_node()))` before this "if"

Reply:
Yeah, I thought about that. But it would look unclear: where is emitter initialization for `ConvertSaturation`. Is there support of `ConvertSaturation` ? To figure out it, you need to go to ctor of `store_emitter` to see that default value of  `arithmetic_mode` is `saturation`.

Because of that, I decided to make debugging easier: explicitly for each `Store` init the corresponding emitter.

Thank you!